src/dll/Hooks/ExecuteProcess.cpp

```diff
@@ -3,30 +3,62 @@
 #include "App.hpp"
 #include "Hook.hpp"
 #include "Systems/ScriptCompilationSystem.hpp"
+#include <winbase.h>
 
 namespace
 {
 bool isAttached = false;
 
-bool _Global_ExecuteProcess(void* a1, RED4ext::CString& aCommand, FixedWString& aArgs,
-                            RED4ext::CString& aCurrentDirectory, char a5);
+bool _Global_ExecuteProcess(ScriptCompilation* a1, RED4ext::CString& aCommand, FixedWString& aArgs,
+                            RED4ext::CString& aCurrentDirectory, ExecuteProcess_Flags aFlags);
 Hook<decltype(&_Global_ExecuteProcess)> Global_ExecuteProcess(Addresses::Global_ExecuteProcess,
                                                               &_Global_ExecuteProcess);
 
-bool _Global_ExecuteProcess(void* a1, RED4ext::CString& aCommand, FixedWString& aArgs,
-                            RED4ext::CString& aCurrentDirectory, char a5)
+bool _Global_ExecuteProcess(ScriptCompilation* a1, RED4ext::CString& aCommand, FixedWString& aArgs,
+                            RED4ext::CString& aCurrentDirectory, ExecuteProcess_Flags aFlags)
 {
     if (strstr(aCommand.c_str(), "scc.exe") == nullptr)
     {
-        return Global_ExecuteProcess(a1, aCommand, aArgs, aCurrentDirectory, a5);
+        return Global_ExecuteProcess(a1, aCommand, aArgs, aCurrentDirectory, aFlags);
     }
 
-    auto scriptCompilationSystem = App::Get()->GetScriptCompilationSystem();
-    
+    auto str = App::Get()->GetScriptCompilationSystem()->GetCompilationArgs(aArgs);
+
     FixedWString newArgs;
-    newArgs.str = scriptCompilationSystem->GetCompilationArgs(aArgs).c_str();
+    newArgs.str = str.c_str();
     newArgs.length = newArgs.maxLength = wcslen(newArgs.str);
-    return Global_ExecuteProcess(a1, aCommand, newArgs, aCurrentDirectory, a5);
+
+    spdlog::info(L"Final redscript compilation arg string: '{}'", newArgs.str);
+    auto result = Global_ExecuteProcess(a1, aCommand, newArgs, aCurrentDirectory, aFlags);
+
+    // 60000 is used in the game
+    auto waitResult = WaitForSingleObject(a1->handle, 60000);
+    switch (waitResult) {
+        case WAIT_TIMEOUT:
+            spdlog::error("Redscript compilation timed-out - exiting");
+            SHOW_LAST_ERROR_MESSAGE_AND_EXIT_FILE_LINE("Redscript compilation timed-out");
+            break;
+        case WAIT_ABANDONED:
+            spdlog::error("Redscript compilation was abandoned - exiting");
+            SHOW_LAST_ERROR_MESSAGE_AND_EXIT_FILE_LINE("Redscript compilation was abandoned");
+            break;
+        case WAIT_FAILED:
+            spdlog::error("Redscript compilation failed - exiting");
+            SHOW_LAST_ERROR_MESSAGE_AND_EXIT_FILE_LINE("Redscript compilation failed");
+            break;
+    }
+
+    GetExitCodeProcess(a1->handle, &a1->errorCode);
```

Comment:
Is this not automatically set when the original is done?

Reply:
These occur in the game after our hook - from what I can tell it's ok to run them twice.